### PR TITLE
Introducing Hashtag + tag indexing

### PR DESCRIPTION
**Pull request by leogrim**
@eishay @andrewconner I have typed the name of a Collection with the value class Hashtag. So that we can get into the habit of passing tags around as typed strings. See also my comments in KeepCommander. 

@ymatsuda tags are available in KeepIndexable, but I haven't spelled out the actual index fields yet (will do next as soon as we merge this). See also my comments in KeepCommander, I'm afraid they were some bugs in the current code for tag search.

I decided not to refactor the current code to inactivate tags together with inactivated keeps (we should) because the "undo" function of the website relies on them. I think we should wait until we get rid of the Collection model to clean this up. 

Thanks @andrewconner for your awesome job on KeepCommander, I agree with you that there's still some work left there :)

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala [hunk 1]
```diff
@@ -571,12 +576,14 @@ class KeepsCommander @Inject() (
       } {
         keepToCollectionRepo.remove(keepId = keep.id.get, collectionId = collection.id.get)
         collectionRepo.collectionChanged(collection.id.get)
+        keepRepo.save(keep) // notify keep index
         keptAnalytics.untaggedPage(collection, keep, context)
       }
     }
     searchClient.updateURIGraph()
   }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala:585]: The method below does not report to analytics (grrrrr)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala [hunk 2]
```diff
@@ -603,6 +611,7 @@ class KeepsCommander @Inject() (
     }
   }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala:613]: The method below does not report to analytics either. It also did not bump up the sequence number of relevant collections, so I'm afraid the current collection index may not be consistent. @ymatsuda

**Comment by ymatsuda** [comment]
:+1:

**Comment by andrewconner** [comment]
Looks good.

**Pull request merged**